ext/intl/rangeformatter/rangeformatter_class.cpp

```diff
@@ -0,0 +1,180 @@
+	zend_object_std_init(&intern->zo, ce);
+	object_properties_init(&intern->zo, ce);
+
+	return &intern->zo;
+}
+
+U_CFUNC PHP_METHOD(IntlNumberRangeFormatter, __construct)
+{
+   
+}
+
+U_CFUNC PHP_METHOD(IntlNumberRangeFormatter, createFromSkeleton)
+{
+    char* skeleton;
+    char* locale;
+    size_t locale_len;
+    size_t skeleton_len;
+    zend_long collapse;
+    zend_long identityFallback;
+
+    ZEND_PARSE_PARAMETERS_START(4,4)
+        Z_PARAM_STRING(skeleton, skeleton_len)
+        Z_PARAM_STRING(locale, locale_len)
+        Z_PARAM_LONG(collapse)
+        Z_PARAM_LONG(identityFallback)
+    ZEND_PARSE_PARAMETERS_END();
+
+    if (locale_len == 0) {
+        locale = (char *)intl_locale_get_default();
+    }
+
+    if (skeleton_len == 0) {
+        zend_argument_value_error(1, "Skeleton string cannot be empty");
+        RETURN_THROWS();
+    }
+
+    if (locale_len > INTL_MAX_LOCALE_LEN) {
+        zend_argument_value_error(2, "Locale string too long, should be no longer than %d characters", INTL_MAX_LOCALE_LEN);
+        RETURN_THROWS();
+    }
+
+    if (strlen(uloc_getISO3Language(locale)) == 0) {
+        zend_argument_value_error(2, "\"%s\" is invalid", locale);
+        RETURN_THROWS();
+    }
+
+    if (collapse != UNUM_RANGE_COLLAPSE_AUTO && collapse != UNUM_RANGE_COLLAPSE_NONE && collapse != UNUM_RANGE_COLLAPSE_UNIT && collapse != UNUM_RANGE_COLLAPSE_ALL) {
+        zend_argument_value_error(3, "Invalid collapse value");
+        RETURN_THROWS();
+    }
+
+    if (identityFallback != UNUM_IDENTITY_FALLBACK_SINGLE_VALUE && identityFallback != UNUM_IDENTITY_FALLBACK_APPROXIMATELY_OR_SINGLE_VALUE && identityFallback != UNUM_IDENTITY_FALLBACK_APPROXIMATELY && identityFallback != UNUM_IDENTITY_FALLBACK_RANGE) {
+        zend_argument_value_error(4, "Invalid identity fallback value");
+        RETURN_THROWS();
+    }
+
+    UErrorCode error = U_ZERO_ERROR;
+
+    UnicodeString skeleton_ustr(skeleton, skeleton_len);
+
+    UnlocalizedNumberFormatter nf = NumberFormatter::forSkeleton(skeleton_ustr, error);
+
+    LocalizedNumberRangeFormatter* nrf = new LocalizedNumberRangeFormatter(
+        NumberRangeFormatter::with()
+            .locale(locale)
+            .numberFormatterBoth(nf)
+            .collapse(static_cast<UNumberRangeCollapse>(collapse))
+            .identityFallback(static_cast<UNumberRangeIdentityFallback>(identityFallback))
+    );
+
+    zend_object* obj = IntlNumberRangeFormatter_object_create(class_entry_IntlNumberRangeFormatter);
+    obj->handlers = &rangeformatter_handlers;
+
+    RANGEFORMATTER_OBJECT(php_intl_numberrangeformatter_fetch_object(obj)) = nrf;
+
+    RETURN_OBJ(obj);
+    
+}
+
+U_CFUNC PHP_METHOD(IntlNumberRangeFormatter, format)
+{
+    double start;
+    double end;
+
+    IntlNumberRangeFormatter_object* obj = Z_INTL_RANGEFORMATTER_P(ZEND_THIS);
+
+    ZEND_PARSE_PARAMETERS_START(2, 2)
+        Z_PARAM_DOUBLE(start)
```

Comment:
this macro implies `float` type, but you would want `int|float` which needs `Z_PARAM_NUMBER`